File: 11726.py
'''
9095 1, 2, 3 더하기랑 비슷함
'''
mem = [None for _ in range(1001)]

mem[1] = 1
mem[2] = 1 * 2

def partition(n: int):
    '''
    2 = 1+1 = 2

    3 = 2+1 = 1+2 = 1+1+1

    4 = 2+2 = 2+1+1 = 1+2+1 = 1+1+2 = 1+1+1+1
    '''
    if n == 1: return mem[1]
    if n == 2: return mem[2]
    if mem[n] is not None: return mem[n]

    mem[n] = partition(n - 1) + partition(n - 2)
    return mem[n]

if __name__ == "__main__":
    N = int(input())
    print(partition(N) % 10007)

File: 11726.java

import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {

    private static final Integer[] mem = new Integer[1001];

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        mem[1] = 1;
        mem[2] = 2;

        int result = partition(N);

        System.out.println(result);
    }

    private static int partition(int n) {
        if (mem[n] != null) {
            return mem[n];
        }

        mem[n] = (partition(n - 1) + partition(n - 2)) % 10007;
        return mem[n];
    }
}
